src/MetadataManager/pathToHash.ts

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as crypto from 'crypto';
+import fs from 'fs';
+import * as vscode from 'vscode';
+
+import {Metadata} from './metadataAPI';
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      await this._instance.validateMetadata(this._instance._map);
+    }
+    return this._instance;
+  }
+
+  private async init() {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const uri = vscode.workspace.workspaceFolders[0].uri;
+
+    return this.getSubPathToHash(uri);
+  }
+
+  private async getSubPathToHash(uri: vscode.Uri) {
+    let subPathToHash: {[key: string]: any} = {};
+    const files = await vscode.workspace.fs.readDirectory(uri);
+
+    for (const file of files) {
+      const name: string = file[0];
+      const type: number = file[1];
+
+      if (type === 1) {
+        subPathToHash[name] = await PathToHash.generateHash(vscode.Uri.joinPath(uri, '/' + name));
+      } else if (type === 2 && name !== '.meta') {
+        subPathToHash[name] = await this.getSubPathToHash(vscode.Uri.joinPath(uri, '/' + name));
+      }
+    }
+
+    return subPathToHash;
+  }
+
+  public async validateMetadata(pathToHash: any) {
```

Comment:
https://github.com/Samsung/ONE-vscode/pull/1288/files#r984465933

And, it uses 'pathToHash' parameter, please use the same name for the argument and parameter. `PathToHash` is used as a class name and its singleton instance, so it can confusing.